サーバが"error: Uncaught (in promise) AssertionError: Assertion failed."で落ちるときがある。

以下の文を吐き出してサーバが落ちます。
```console
error: Uncaught (in promise) AssertionError: Assertion failed.
    clearInterval(this.intervalId);
    ^
    at assert (deno:runtime/js/06_util.js:34:13)
    at setGlobalTimeout (deno:runtime/js/11_timers.js:317:5)
    at setOrClearGlobalTimeout (deno:runtime/js/11_timers.js:362:7)
    at unschedule (deno:runtime/js/11_timers.js:411:9)
    at clearTimer (deno:runtime/js/11_timers.js:533:5)
    at clearInterval (deno:runtime/js/11_timers.js:550:5)
    at ExpGame.dispose (file:///C:/Users/xxxxxxxxxxxx/Kakomimasu/apiserver/parts/expKakomimasu.js:89:5)
    at ExpGame.updateStatus (file:///C:/Users/xxxxxxxxxxxx/Kakomimasu/apiserver/parts/expKakomimasu.js:71:12)
    at file:///C:/Users/xxxxxxxxxxxx/Kakomimasu/apiserver/parts/expKakomimasu.js:25:48
    at fire (deno:runtime/js/11_timers.js:446:7)
```

クライアントを大量に起動した際にまれに起きることがあります。（つまり再現不可）

--- a/apiserver/apiserver.ts
+++ b/apiserver/apiserver.ts
@@ -315,8 +315,6 @@ export const getGameInfo = async (req: ServerRequest) => {
     ]
       .find((item) => item.uuid === id);
     if (game) {
-      game.updateStatus();
-
       await req.respond({
         status: 200,
         headers: new Headers({

--- a/apiserver/parts/expKakomimasu.js
+++ b/apiserver/parts/expKakomimasu.js
@@ -49,18 +49,17 @@ class ExpGame extends Game {
   }
 
   updateStatus() {
-    let self = this;
     if (
-      self.isReady() && !self.isGaming() && !self.ending &&
+      this.isReady() && !this.isGaming() && !this.ending &&
       (new Date().getTime() > (this.startedAtUnixTime * 1000))
     ) {
-      self.start();
-      this.wsSend(); //this.changeFuncs.forEach((func) => func());
+      this.start();
+      this.wsSend();
     }
-    if (self.isGaming()) {
+    if (this.isGaming()) {
       if (new Date().getTime() > (this.nextTurnUnixTime * 1000)) {
-        self.nextTurn();
-        this.wsSend(); //this.changeFuncs.forEach((func) => func());
+        this.nextTurn();
+        this.wsSend();
       }
     }
     if (this.ending) {
@@ -69,7 +68,7 @@ class ExpGame extends Game {
 
       this.dispose();
       console.log("turn", this.turn);
-      this.wsSend(); //this.changeFuncs.forEach(func => func());
+      this.wsSend();
     }
   }
 

--- a/apiserver/parts/expKakomimasu.js
+++ b/apiserver/parts/expKakomimasu.js
@@ -22,10 +22,8 @@ class ExpGame extends Game {
       this.startedAtUnixTime = Math.floor(new Date().getTime() / 1000) + 5;
       this.nextTurnUnixTime = this.startedAtUnixTime + this.nsec;
       this.updateStatus();
-      this.intervalId = setInterval(() => this.updateStatus(), 50);
-      //console.log("intervalID", this.intervalId);
     }
-    this.wsSend(); //this.changeFuncs.forEach((func) => func());
+    this.wsSend();
     return true;
   }
 
@@ -49,26 +47,31 @@ class ExpGame extends Game {
   }
 
   updateStatus() {
-    if (
-      this.isReady() && !this.isGaming() && !this.ending &&
-      (new Date().getTime() > (this.startedAtUnixTime * 1000))
-    ) {
-      this.start();
-      this.wsSend();
-    }
-    if (this.isGaming()) {
-      if (new Date().getTime() > (this.nextTurnUnixTime * 1000)) {
+    if (this.isGaming()) { // ゲーム進行中
+      const diff = (this.nextTurnUnixTime * 1000) - new Date().getTime();
+      if (diff <= 0) {
         this.nextTurn();
         this.wsSend();
+        this.updateStatus();
+      } else {
+        setTimeout(() => this.updateStatus(), diff);
       }
     }
-    if (this.ending) {
-      //saveLogFile(this);
+    else if (this.ending) { // ゲーム終了後
       LogFileOp.save(this);
 
-      this.dispose();
       console.log("turn", this.turn);
       this.wsSend();
+    } // ゲーム開始前
+    else {
+      const diff = (this.startedAtUnixTime * 1000) - new Date().getTime();
+      if (diff <= 0) {
+        this.start();
+        this.wsSend();
+        this.updateStatus();
+      } else {
+        setTimeout(() => this.updateStatus(), diff);
+      }
     }
   }
 
@@ -81,11 +84,6 @@ class ExpGame extends Game {
       nextTurnUnixTime: this.nextTurnUnixTime,
       reservedUsers: this.reservedUsers,
     };
-  }
-
-  dispose() {
-    //console.log(this.intervalId);
-    clearInterval(this.intervalId);
   }
 
   wsSend() {

